govcd/nsxt_tier0_router.go

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2020 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// NsxtTier0Router
+type NsxtTier0Router struct {
+	NsxtTier0Router *types.NsxtTier0Router
+	client          *Client
+}
+
+// GetImportableNsxtTier0RouterByName retrieves NSX-T tier 0 router by given parent NSX-T manager ID and Tier 0 router
+// name
+//
+// Warning. The API returns only unused Tier-0 routers (the ones that are not used in external networks yet)
+//
+// Note. NSX-T manager ID is mandatory and must be in URN format (e.g.
+// urn:vcloud:nsxtmanager:09722307-aee0-4623-af95-7f8e577c9ebc)
+
+func (vcdCli *VCDClient) GetImportableNsxtTier0RouterByName(name, nsxtManagerId string) (*NsxtTier0Router, error) {
+	if nsxtManagerId == "" {
+		return nil, fmt.Errorf("no NSX-T manager ID specified")
+	}
+
+	if !isUrn(nsxtManagerId) {
+		return nil, fmt.Errorf("NSX-T manager ID is not URN (e.g. 'urn:vcloud:nsxtmanager:09722307-aee0-4623-af95-7f8e577c9ebc)', got: %s", nsxtManagerId)
+	}
+
+	if name == "" {
+		return nil, fmt.Errorf("empty Tier 0 router name specified")
+	}
+
+	// Ideally FIQL filter could be used to filter on server side and get only desired result, but filtering on
+	// 'displayName' is not yet supported. The only supported field for filtering is
+	// _context==urn:vcloud:nsxtmanager:09722307-aee0-4623-af95-7f8e577c9ebc to specify parent NSX-T manager (This
+	// automatically happens in GetAllImportableNsxtTier0Routers()). The below filter injection is left as documentation.
+	/*
+		queryParameters := copyOrNewUrlValues(nil)
+		queryParameters.Add("filter", "displayName=="+name)
+	*/
+
+	nsxtTier0Routers, err := vcdCli.GetAllImportableNsxtTier0Routers(nsxtManagerId, nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not find NSX-T Tier-0 router with name '%s' for NSX-T manager with id '%s': %s",
+			name, nsxtManagerId, err)
+	}
+
+	// TODO remove this when FIQL supports filtering on 'displayName'
```

Comment:
maybe better is move this part to private func and deprecate it :)

Reply:
Moved to a function.